code/modules/mob/living/carbon/human/species.dm

```diff
@@ -331,6 +333,13 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			C.dropItemToGround(thing)
 	if(C.hud_used)
 		C.hud_used.update_locked_slots()
+	///Cycles through all their equipped items and unequips any they arent supposed to wear according to the new restriction system.
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		for(var/obj/item/I in H.get_equipped_items(TRUE))
+			if(I.species_restricted & H.dna.species.bodyflag)
+				H.dropItemToGround(I)
```

Comment:
There is proc for that

Reply:
Thanks for the *detailed* suggestion, Arch